internal/scheduler/metrics/metrics.go

```diff
@@ -383,7 +441,7 @@ func (m *Metrics) updateCounterVecFromJob(vec *prometheus.CounterVec, labels []s
 		labels[i] = string(resourceName)
 		if c, err := vec.GetMetricWithLabelValues(labels...); err != nil {
 			return err
-		} else {
+		} else if len(labels) == cap(labels) { // We consume all the buffer, therefore it
```

Comment:
Is this needed? Comparing against the cap seems fragile.

Reply:
I was thinking of something, and forgot to delete it